utils/gitlog2changelog.py

```diff
@@ -4,18 +4,31 @@
 # Distributed under the terms of the GNU General Public License v2 or later
 
 import re
-import os
 from textwrap import TextWrapper
 import sys
+import subprocess
 
 rev_range = ""
 
 if len(sys.argv) > 1:
     base = sys.argv[1]
     rev_range = "%s..HEAD" % base
 
+# Define the git command and its arguments as a list
+git_command = [
+    "git",
+    "log",
+    "--summary",
+    "--stat",
+    "--no-merges",
+    "--date=short",
+    rev_range,
```

Comment:
This is not the same as the previous code. If you [download the artifact](https://github.com/OSGeo/grass/actions/runs/8347841214), you will see that the ChangeLog file this is generating is empty. The error is in the [build log](https://github.com/OSGeo/grass/actions/runs/8347841214/job/22849215881). It comes from Git. Putting an empty string at the end of a string in the original os.popen code is not the same as providing an empty argument in the new code. In Bash, it is a difference between these two:

```bash
git log --summary --stat --no-merges --date=short
git log --summary --stat --no-merges --date=short ""
```

rev_range needs to be optionally appended.

Reply:
thanks for this comment, It helped in debugging a lot. optional appending was added in the latest version.

Reply:
Artifact now contains a ChangeLog file. Its content looks good!